autowiring/auto_signal.h

```diff
@@ -201,7 +224,9 @@ namespace autowiring {
     typedef internal::signal_node<Args...> registration_t;
     typedef std::function<void(Args...)> function_t;
 
-    registration_t* operator+=(function_t&& fn) { return *m_relay += std::move(fn); }
+    template<typename t_Fn>
+    registration_t* operator+=(t_Fn && fn) { return *m_relay += std::move(fn); }
```

Comment:
Use `std::forward`, here, not `std::move`.  You can't know for sure that `t_Fn` is an rvalue reference because of rvalue reference collapse rules.